pybamm/solvers/jax_bdf_solver.py

```diff
@@ -26,6 +27,45 @@
 MAX_FACTOR = 10
 
 
+# https://github.com/google/jax/issues/4572#issuecomment-709809897
+def some_hash_function(x):
+    return int(onp.sum(x))
```

Comment:
an alternative is to use `to_bytes`, see https://github.com/pybamm-team/PyBaMM/blob/e025cedfd3de4d1185d88adf3c7e22921bcca3ae/pybamm/expression_tree/array.py#L70-L86

Reply:
yea, good point. I think `tobytes` is better, so swapped to that.